File: brain/71051363-75eb-4ceb-9201-a081e9878f8c/scratch/read_exif.js
const fs = require('fs');

const file = 'c:\\Users\\BLUEWAVE COMP\\Documents\\projects\\ambience-vista\\client\\src\\assets\\4dd8bc2c-07e0-4e5e-bf29-9e96c1ad6230.jpg';
if (!fs.existsSync(file)) {
  console.log('File does not exist');
  process.exit(1);
}

const buffer = fs.readFileSync(file);
console.log('Reading first 2000 bytes for strings...');
const printable = [];
for (let i = 0; i < Math.min(buffer.length, 10000); i++) {
  const char = buffer[i];
  if (char >= 32 && char <= 126) {
    printable.push(String.fromCharCode(char));
  } else {
    printable.push(' ');
  }
}
const text = printable.join('');
// Find runs of text
const matches = text.match(/[A-Za-z0-9\s\-\/\:\.]{4,}/g);
if (matches) {
  console.log('Found metadata strings:');
  const filtered = matches.map(m => m.trim()).filter(m => m.length > 3);
  console.log(filtered.slice(0, 30));
} else {
  console.log('No metadata strings found');
}

File: brain/71051363-75eb-4ceb-9201-a081e9878f8c/scratch/read_exif.py
import os
from PIL import Image
from PIL.ExifTags import TAGS

file_path = r"c:\Users\BLUEWAVE COMP\Documents\projects\ambience-vista\client\src\assets\IMG_6309.jpg"

if not os.path.exists(file_path):
    print("File does not exist")
    sys.exit(1)

try:
    image = Image.open(file_path)
    exif_data = image._getexif()
    if exif_data:
        print("EXIF Tags found:")
        for tag, value in exif_data.items():
            tag_name = TAGS.get(tag, tag)
            # Skip large binary data tags (like MakerNote)
            if tag_name not in ['MakerNote', 'UserComment']:
                print(f"  {tag_name}: {value}")
    else:
        print("No EXIF data found in image")
except Exception as e:
    print("Error reading EXIF:", e)
